Readme.md

```diff
@@ -64,37 +70,62 @@ or
-  * `S3` - Finally, if this is chosen, the message will be stored directly as a file into `s3BucketName`. No SQS message is send.
+* `queueName` - this is the name of an SQS normal queue to be used for messages which `length < maxMessageBytes`.
+  Discarded & truncated message will be included as well.
+* `largeMessageMode` - select one of the provided options `(TRUNCATE|DISCARD|FIFO|EXTENDED|S3)` to process an event message larger than `maxMessageBytes`
+  before sent it to target.
+    * `TRUNCATE` - message will be truncated to `maxMessageBytes`.
+    * `DISCARD` - It will modify message by changing original content
+      to `Can't display log message. Message max size exceeded. max size: maxMessageBytes`.
+    * `FIFO` - this option will split the messages into multiple SQS messages and placed on a FIFO queue with the same
+      group ID value (the uuid) so they are guaranteed to be processed in order. The SQS messages will need to be
+      accumulated so the log message can be reconstructed.
+      The message format looks like
+      this: `currentPart=1|totalParts=5|uuid=8744fe2c-cbea-4e49-8b81-0d7076899a48|message="[INFO] - Hello World! "` 
+    * `EXTENDED` - In this case the message will be queued into SQS `queueName` queue with a reference id to `s3BucketName`
+      bucket file, which at the end, will contain the actual message.
+    * `S3` - Finally, if this is chosen, the message will be stored directly as a file into `s3BucketName`. No SQS
+      message is send.
 
 Example with all possible configurations:
+
 ```xml
 <Appenders>
     <SQS name="SQS" 
-         awsAccessKey="ACCESSKEY"
-         awsRegion="us-east-1"
-         awsSecretKey="SECRETKEY"
+         awsAccessKey="YOURACCESSKEY"
+         awsRegion="YOURAWSREGION"
+         awsSecretKey="YOURSECRETKEY"
          maxBatchOpenMs="10000"
          maxBatchSize="1"
          maxInflightOutboundBatches="1"
-         queueName="<a normal length message queue name>"
-         largeMessageMode="FIFO"
-         largeMessageQueueName="<a large length message queue name.fifo>"
-         s3BucketName="<a s3 bucket name>"
+         largeMessageMode="EXTENDED"
+         queueName="stg-normal-messages-queue"
+         largeMessageQueueName="stg-large-messages-queue"
+         s3BucketName="stg-sqs-messages-bucket"
          maxMessageBytes="256">
          <PatternLayout pattern="%-5p %d [%t] %c: ##MESSAGE## %m%n"/>
     </SQS>
 </Appenders>
 ```
+Run Automated tests
+==========================
+* Run automated test of this componen using the following command.
+
+```
+mvn clean test -DawsAccessKey=YOURAWSKEY /
+ -DawsSecretKey=YOURAWSSECRET / 
+ -DawsRegion=YOURAWSREGION -DawsBucketName=your-bucket /
+ -DawsQueueName=your-queue / 
+ -DawsLargeMessageQueueName=your-large-queue /
+ -DmaxMessageBytes=25600
```

Comment:
Also 25600 needs to be 256 otherwise the test does not pass

Reply:
Done